src/test/java/build/buildfarm/instance/memory/ByteStringStreamSourceTest.java

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.instance.memory;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.io.InputStream;
+
+@RunWith(JUnit4.class)
+public class ByteStringStreamSourceTest {
+  @Test
+  public void closeCallsOnClose() throws IOException {
+    Runnable mockOnClose = mock(Runnable.class);
+    ByteStringStreamSource source = new ByteStringStreamSource(mockOnClose);
+    verify(mockOnClose, never()).run();
+    source.getOutputStream().close();
+    verify(mockOnClose).run();
```

Comment:
nit: verify(mockOnClose, times(1)).run();
And ditto in all other cases. I think the default verifies at least once, not exactly once.